specification/elasticsan/resource-manager/Microsoft.ElasticSan/preview/2022-12-01-preview/elasticsan.json

```diff
@@ -1008,36 +1038,219 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ElasticSan/elasticSans/{elasticSanName}/privateEndpointConnections/{privateEndpointConnectionName}": {
+      "put": {
+        "tags": [
+          "PrivateEndpointConnections"
+        ],
+        "description": "Update the state of specified private endpoint connection associated with the Elastic San",
+        "operationId": "PrivateEndpointConnections_Create",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ElasticSanNameParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateEndpointConnectionNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            },
+            "description": "Private Endpoint Connection Approval object."
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Returned with the operation is successfully completed.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            }
+          },
+          "201": {
+            "description": "Created -- Resource Created; operation will complete asynchronously",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            }
+          },
+          "default": {
+            "description": "Error from Elastic SAN resource provider.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
```

Comment:
I see some existing API has changed it to `"final-state-via": "location"`. 
should we also change it in this line?

Reply:
I got this change from previous swagger PR ( pipeline was failing), can you please explain the impact of this, and why is it needed?

Reply:
@kazrael2119

Could you please give some details for the difference of  `"final-state-via": "location"` and `"final-state-via": "azure-async-operation"`

Reply:
@blueww sorry, I have not any knowledge about this, @raych1 can you give some details for this? thanks

Reply:
@blueww , please refer to http://azure.github.io/autorest/extensions/#x-ms-long-running-operation-options. The value should align with service implementation.

Reply:
Thanks @raych1 !

@PreetPRG 
Would you please review the doc link above, and make sure swagger is aligned with service implementation?

Reply:
Thanks @blueww , i went through doc and feels location makes more sense. Updated.